### ISSUE TITLE
PSKX model is broken upon import

### ISSUE THREAD

**Issue by 6yXJI0** [DarklightGames/io_scene_psk_psa#43]
![image](https://github.com/DarklightGames/io_scene_psk_psa/assets/32490288/eda62c0f-dfb0-414c-91d9-a6f827e2ca6d)
[Johnny_PSKX.zip](https://github.com/DarklightGames/io_scene_psk_psa/files/12297185/Johnny_PSKX.zip)
```
PSK imported with 1 warning(s)
Discarded 561 degenerate face(s).
```

**Comment by cmbasnett** [comment]
Thanks for the report,  I will look into this. What game is this from?

**Comment by 6yXJI0** [comment]
> Thanks for the report, I will look into this. What game is this from?

Texas Chain Saw Massacre (Tech Test Build)

**Comment by cmbasnett** [comment]
I managed to fix it on my end, but I'll need to do a bit of testing to make sure the fix isn't going to break anything else.

![image](https://github.com/DarklightGames/io_scene_psk_psa/assets/5035660/6018afde-32af-412c-a4ee-d7177040c466)

The issue was that the mesh has more than 65536 "wedges" or "face vertices". The specification sheet for the format I was working with indicated that the wedges were only 16-bit addressable (with 16 bits of "padding" after it), but it seems that for this file at least, it actually uses the full 32-bits. Flipping wedge index to be a 32-bit integer fixes the issue.

I will run a few tests locally here, but I suspect this will be an easy fix and I can publish a hotfix tonight.

**Comment by cmbasnett** [comment]
@6yXJI0 This has been fixed and is available in draft release 5.0.2 (https://github.com/DarklightGames/io_scene_psk_psa/releases/tag/5.0.2). Please re-install the addon with this new version and let me know that it is solved on your end.

**Comment by 6yXJI0** [comment]
> @6yXJI0 This has been fixed and is available in draft release 5.0.2 (https://github.com/DarklightGames/io_scene_psk_psa/releases/tag/5.0.2). Please re-install the addon with this new version and let me know if this solves your issue!
![NVIDIA_Share_ikBXmTozB1](https://github.com/DarklightGames/io_scene_psk_psa/assets/32490288/54881177-f213-43d5-852a-e2f96cb5e4e7)

![NVIDIA_Share_jtRQ6z1OIC](https://github.com/DarklightGames/io_scene_psk_psa/assets/32490288/c06fd49c-d4f6-4789-9689-0f9c6ada0787)

![NVIDIA_Share_BrmujdJwIf](https://github.com/DarklightGames/io_scene_psk_psa/assets/32490288/f92ab40c-c14c-4fc5-8440-adf9765ac62c)

Yes! Thank you so much!

**Comment by cmbasnett** [comment]
The problem, I believe, is that there are duplicate faces on the mesh (i.e., a face consisting of the same 3 vertices), and Blender does not allow this for technical reasons (probably having to do with hashing of the faces).

A possible solution for this would be to split the PSK into as many different meshes as is needed in order to have all of the faces represented in Blender. I will make a new task to sort this out.
